tests/e2e/tests/multicluster/split_horizon_test.go

```diff
@@ -0,0 +1,404 @@
+	c.removed = []string{}
+	for _, r := range c.Removes {
+		content, err := util.KubeGetYaml(c.Namespace, r.Kind, r.Name, c.kubeconfig)
+		if err != nil {
+			// Run the teardown function now and return
+			_ = c.Teardown()
+			return err
+		}
+		if err := util.KubeDeleteContents(c.Namespace, content, c.kubeconfig); err != nil {
+			// Run the teardown function now and return
+			_ = c.Teardown()
+			return err
+		}
+		c.removed = append(c.removed, content)
+	}
+
+	if c.Namespace != tc.Kube.Namespace {
+		if err := util.CreateNamespace(c.Namespace, c.kubeconfig); err != nil {
+			_ = c.Teardown()
+			return err
+		}
+	}
+
+	if c.IstioInjection {
+		if err := util.LabelNamespace(c.Namespace, "istio-injection=enabled", c.kubeconfig); err != nil {
+			_ = c.Teardown()
+			return err
+		}
+	}
+
+	c.applied = []string{}
+	// Apply the yaml file(s)
+	for _, yamlFile := range c.YamlFiles {
+		kubeCmd := "apply"
+		if len(yamlFile.LabelSeletor) > 0 {
+			kubeCmd += " -l " + yamlFile.LabelSeletor
+		}
+		if err := util.KubeCommand(kubeCmd, c.Namespace, yamlFile.Name, c.kubeconfig); err != nil {
+			// Run the teardown function now and return
+			_ = c.Teardown()
+			return err
+		}
+		c.applied = append(c.applied, yamlFile.Name)
+	}
+
+	// Wait until the deployments are ready
+	if err := util.CheckDeployments(c.Namespace, maxDeploymentTimeout, c.kubeconfig); err != nil {
+		_ = c.Teardown()
+		return err
+	}
+
+	return nil
+}
+
+// Teardown deletes the deployed configuration.
+func (c *deployableConfig) Teardown() error {
+	err := c.TeardownNoDelay()
+
+	// Sleep for a while to allow the change to propagate.
+	time.Sleep(c.propagationDelay())
+	return err
+}
+
+// Teardown deletes the deployed configuration.
+func (c *deployableConfig) TeardownNoDelay() error {
+	var err error
+	for _, yamlFile := range c.applied {
+		err = multierr.Append(err, util.KubeDelete(c.Namespace, yamlFile, c.kubeconfig))
+	}
+	// Restore configs that was removed
+	for _, yaml := range c.removed {
+		err = multierr.Append(err, util.KubeApplyContents(c.Namespace, yaml, c.kubeconfig))
+	}
+	c.applied = []string{}
+	return err
+}
+
+func (c *deployableConfig) propagationDelay() time.Duration {
+	// With multiple clusters, it takes more time to propagate.
+	return defaultPropagationDelay * time.Duration(len(tc.Kube.Clusters))
+}
+
+type testConfig struct {
+	*framework.CommonConfig
+	AppDir      string
+	extraConfig map[string]*deployableConfig
+}
+
+// Setup initializes the test environment and waits for all pods to be in the running state.
+func (t *testConfig) Setup() (err error) {
+	// Wait for all the pods to be in the running state before starting tests.
+	for cluster, kc := range t.Kube.Clusters {
+		log.Infof("Making sure all pods are running on cluster: %s", cluster)
+		if err == nil && !util.CheckPodsRunning(t.Kube.Namespace, kc) {
+			err = fmt.Errorf("can't get all pods running in %s cluster", cluster)
+			break
+		}
+		log.Info("All pods are running")
+	}
+
```

Comment:
You should consider adding calls to verifyMeshConfig and/or createAndVerifyMCMeshConfig to verify that the mesh configuration learned from the remote cluster is correct

Reply:
It relates to https://github.com/istio/istio/pull/12709#issuecomment-480568094. verifyMeshConfig() and createAndVerifyMCMeshConfig() should be part of the test of type 1, similarly to the current multicluster test. That test can be submitted as a separate PR.

This PR submits a different type of a multi-cluster test - testing a specific scenario (documented on istio.io) with cross-cluster communication.